projects/packages/forms/src/contact-form/class-contact-form-field.php

```diff
@@ -1391,23 +1402,80 @@ public function render_outline_label( $id, $label, $required, $required_field_te
 		$classes .= $this->is_error() ? ' form-error' : '';
 		$classes .= $this->label_classes ? ' ' . $this->label_classes : '';
 
+		$output_data = $this->get_outline_styles();
+
 		return '
 			<div class="notched-label">
-				<div class="notched-label__leading"></div>
-				<div class="notched-label__notch">
+				<div class="notched-label__leading ' . $output_data['class_name'] . '"' . $output_data['style'] . '></div>
+				<div class="notched-label__notch ' . $output_data['class_name'] . '"' . $output_data['style'] . '>
 					<label
 						for="' . esc_attr( $id ) . '"
 						class=" ' . $classes . '"
-						style="' . $this->label_styles . '"
+						style="' . $this->label_styles . $output_data['css_vars'] . '"
 					>'
 			. esc_html( $label )
 			. ( $required ? '<span class="grunion-label-required" aria-hidden="true">' . $required_field_text . '</span>' : '' ) .
 			'</label>
 				</div>
-				<div class="notched-label__trailing"></div>
+				<div class="notched-label__trailing ' . $output_data['class_name'] . '"' . $output_data['style'] . '></div>
 			</div>';
 	}
 
+	/**
+	 * Returns the styles, classes and CSS variables associated with the outline style.
+	 *
+	 * @return array {
+	 *     @type string $style_attrs The style attributes.
+	 *     @type string $css_vars The CSS variables.
+	 *     @type string $class_name The class name.
+	 * }
+	 */
+	private function get_outline_styles() {
+		$style_attrs     = '';
+		$css_vars        = '';
+		$outline_styles  = $this->get_attribute( 'outlinestyledata' );
+		$outline_classes = $this->get_attribute( 'outlinestyleclasses' );
+
+		if ( ! empty( $outline_styles ) ) {
+			$outline_styles = json_decode( html_entity_decode( $outline_styles, ENT_COMPAT ), true );
+			$input_styles   = $this->get_attribute( 'inputstyles' );
+
+			if ( ! empty( $input_styles ) ) {
+				$style_attrs = ' style="' . esc_attr( $input_styles ) . '"';
+			}
+
+			$global_styles = wp_get_global_styles(
+				array( 'border' ),
+				array(
+					'block_name' => 'jetpack/input',
+					'transforms' => array( 'resolve-variables' ),
+				)
+			);
+
+			$legacy_border_size   = ( isset( $outline_styles['width'] ) && is_numeric( $outline_styles['width'] ) ) ? $outline_styles['width'] . 'px' : $outline_styles['width'];
```

Comment:
The linter is complaining about this line and the one below because the keys `width` and `border` might not exist. Could something like the following work instead?

```php
			$legacy_border_size   = $outline_styles['width'] ?? null;
			$legacy_border_size   = is_numeric( $legacy_border_size ) ? $legacy_border_size . 'px' : $legacy_border_size;
			$legacy_border_radius = $outline_styles['radius'] ?? null;
			$legacy_border_radius = is_numeric( $legacy_border_radius ) ? $legacy_border_radius . 'px' : $legacy_border_radius;
```

Reply:
Strange. The linter can't recognise the first `isset( $outline_styles['width'] )`?

Reply:
No, I think the problem is that the second operand of the ternary falls back to `$outline_styles['width']` which sits outside of the `( isset( $outline_styles['width'] ) && is_numeric( $outline_styles['width'] ) )` check. I.e. if that evaluates to `false` we attempt to grab `['width']` anyway, even though it might not be set.

Reply:
Done, thanks for the example code too. 🙇🏻